pkg/ephemeral/io/carrier.go

```diff
@@ -42,16 +46,47 @@ type Config struct {
 }
 
 // Connect establishes a TCP connection to a socket on a given host and port.
-func (c *Carrier) Connect(ctx context.Context, host, port string) error {
+func (c *Carrier) Connect(ctx context.Context, playerID int32, host string, port string) error {
 	conn, err := c.Dialer(ctx, host, port)
 	if err != nil {
 		return err
 	}
-	c.Conn = conn
+	_, err = conn.Write(c.buildHeader(playerID))
```

Comment:
Just to be sure that this is correct: First, we write the header and then we do the TLS upgrade?

Reply:
Yes, first we tell MP-SPDZ which Player-ID we are, then we upgrade the connection